### PR TITLE
Redux react

### PR DESCRIPTION
**Pull request by franclobo**
### Use Redux in React components.

- Keep state immutable in reducers!
- Change the default state in the books reducer from an empty array to an array with a few books. One of the goals is to display them in your React components.
- Import Redux Provider and the store in the main component of your React app.
- Connect the app to the Redux store.
- The list of books displays the books from Redux store.
- Use useSelector to consume the state.

**In React component responsible for adding new books, add functionality for the "Add new" button click:**

- Import Redux useDispatch.
- Set the data inputs in the local React state (set title and author - NOTE: categories and comments are NOT part of this step).
- Once the new book object is ready to be submitted to the Redux store, dispatch a corresponding action (import Action Creator that you created before and use it here).

**In the React component responsible for removing books, add functionality for the "Remove" button click:**

- Import Redux useDispatch.
- Dispatch a corresponding action (import Action Creator that you created before and use it here).

**In your React component responsible for categories, add functionality for the "Check status" button click:**

- Import Redux useDispatch.
- Dispatch a corresponding action (import Action Creator that you created before and use it here).

### CODE REVIEW

File: src/components/Book.js [hunk 1]
```diff
@@ -1,30 +1,30 @@
...
-        </li>
+        {
+          listBooks.map((item) => (
+            <li key={item.id}>
+              {item.title}
+              <br />
+              {item.author}
+              <br />
+              <button className="removeBtn" type="button" onClick={() => dispatch(removeBook())} key={item.id}>Remove</button>
+            </li>
```
Comment (DammyShittu) [inline src/components/Book.js]: - [x] There should be another component, that renders a single book. Kindly create a new reusable component that represents a book, pass in `title` and `author` as props and render them in that component.

  Example
  ```
   <SingleBook title={item.title} author={item.author}>
  ```
  Then in the SingleBook component you can render this code accordingly.

File: src/components/Book.js [hunk 2]
```diff
@@ -1,30 +1,30 @@
...
+import { useSelector, useDispatch } from 'react-redux';
+import { removeBook } from '../redux/books/Books';
 import InputBook from './Form';
 
-function Book(props) {
-  const { title, author } = props;
+function Book() {
+  const dispatch = useDispatch();
+  const listBooks = useSelector((store) => store.books);
+  console.log(listBooks);
```
Comment (DammyShittu) [inline src/components/Book.js]: - [x] Kindly remove unnecessary console log here. It is not good practice to push them and it kills performance

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @franclobo,

✔️ Book can be added and removed

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
- [x] The button in the categories component should dispatch an action to the CategoriesReducer that returns "Under construction". Kindly do that and set up the categories actions and reducer accordingly

  categ-1

_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mateo951 — APPROVED** [review]
### Hi @franclobo, 

Well done making the changes requested by the previous reviewer 🚀
Your project is complete!  There is nothing else to say other than... it's time to merge it 🤓 
### Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**